### PR TITLE
`--format` and `--filter` options for `network ls` and `network inspect` command

### PR DESCRIPTION
**Pull request by kunalkushwaha**
This PR is for fixing bug #5205 
This adds following options to network commands 
  - `podman network ls --filter` option 
  -  `podman network inspect --format` option

NOTE: _Since the CNI conf file does not have defined structure and is read as read & stored as `map of interfaces`, its difficult to get similar output as of `docker` as specified in https://github.com/containers/libpod/issues/5205#issue-565036875. until we create an internal structure. Which may not be good idea, as it may require to sync up with libcni code. Ideally, this structure should be defined in libcni code._

PTAL: @medyagh @afbjorklund, If this helps your use case.

### CODE REVIEW

**Comment by sujil02** [comment]
It might be a good idea to add some test cases here https://github.com/containers/libpod/blob/master/test/e2e/network_test.go#L95 corresponding to new flags.

**Comment by TomSweeneyRedHat** [comment]
Quick peruse LGTM.  Need tests and also updates to `./completions/bash/podman`

File: docs/source/markdown/podman-network-ls.1.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ Displays a list of existing podman networks. This command is not available for r
 
 The `quiet` option will restrict the output to only the network names
 
+**--format**, **-f**
+
+Pretty-print networks to JSON or using a Go template
+
+**--fiter**
```
Comment (rhatdan) [inline docs/source/markdown/podman-network-ls.1.md:21]: Spelling.

File: docs/source/markdown/podman-network-inspect.1.md [hunk 2]
```diff
@@ -9,6 +9,15 @@ podman\-network\-inspect - Displays the raw CNI network configuration for one or
...
 Display the raw (JSON format) network configuration. This command is not available for rootless users.
 
+## OPTIONS
+**--quiet**, **-q**
+
+The `quiet` option will restrict the output to only the network names
+
+**--format**, **-f**
+
+Pretty-print networks to JSON or using a Go template
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-network-inspect.1.md:19]: ending period

File: docs/source/markdown/podman-network-inspect.1.md [hunk 3]
```diff
@@ -9,6 +9,15 @@ podman\-network\-inspect - Displays the raw CNI network configuration for one or
 ## DESCRIPTION
 Display the raw (JSON format) network configuration. This command is not available for rootless users.
 
+## OPTIONS
+**--quiet**, **-q**
+
+The `quiet` option will restrict the output to only the network names
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-network-inspect.1.md:15]: ending period

File: docs/source/markdown/podman-network-ls.1.md [hunk 4]
```diff
@@ -14,6 +14,14 @@ Displays a list of existing podman networks. This command is not available for r
 
 The `quiet` option will restrict the output to only the network names
 
+**--format**, **-f**
+
+Pretty-print networks to JSON or using a Go template
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-network-ls.1.md:19]: ending period

File: docs/source/markdown/podman-network-ls.1.md [hunk 5]
```diff
@@ -14,6 +14,14 @@ Displays a list of existing podman networks. This command is not available for r
...
+
+**--filter**
+
+Provide filter values (e.g. 'name=podman')
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-network-ls.1.md:23]: ending period

File: cmd/podman/networks/inspect.go [hunk 6]
```diff
@@ -41,6 +51,23 @@ func networkInspect(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	fmt.Println(string(b))
+	if strings.ToLower(networkInspectOptions.Format) == "json" || networkInspectOptions.Format == "" {
+		fmt.Println(string(b))
+	} else {
+
```
Comment (QiWang19) [inline cmd/podman/networks/inspect.go:57]: empty line not needed

**Comment by sujil02** [comment]
@rhatdan: Need further opinion. I might be wrong.

Generally from what I have seen filters are usually applied lower on the stack bypassing the filter as params down. From how its done here filters for the network cannot be used by remote clients resulting in ambiguity in the responses. This can be considered a workaround in my opinion.

**Comment by rhatdan** [comment]
@sujil02 Not sure what you are saying, But I believe the filtering should be taking place on the server side not in the client.

**Comment by sujil02** [comment]
What I meant was the api for network list will not be able to implement filters as it is not bonded to the `url.values()` and passed down.

**Comment by QiWang19** [comment]
yes, I think this filter should be passing down. With the tunnel mode NetworkList hasn't been implemented yet, maybe for this PR is should pass to pkg/domain/infra/abi/network.go?

**Comment by kunalkushwaha** [comment]
Got it. I will move the filter code from cli to `pkg`

**Comment by QiWang19** [comment]
LGTM

**Pull request merged**